GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -625,9 +625,31 @@ - (void)addSaveAuthCallback:(GIDAuthFlow *)authFlow {
                                                  code:kGIDSignInErrorCodeKeychain];
         return;
       }
-      GIDGoogleUser *user = [[GIDGoogleUser alloc] initWithAuthState:authState
-                                                         profileData:handlerAuthFlow.profileData];
-      [self setCurrentUserWithKVO:user];
+
+      NSArray<NSString *> *grantedScopes;
+      NSString *grantedScopeString = authState.lastTokenResponse.scope;
+      if (grantedScopeString) {
+        // If we have a 'scope' parameter from the backend, this is authoritative.
+        // Remove leading and trailing whitespace.
+        grantedScopeString = [grantedScopeString stringByTrimmingCharactersInSet:
+            [NSCharacterSet whitespaceCharacterSet]];
+        // Tokenize with space as a delimiter.
+        NSMutableArray<NSString *> *parsedScopes =
+            [[grantedScopeString componentsSeparatedByString:@" "] mutableCopy];
+        // Remove empty strings.
+        [parsedScopes removeObject:@""];
+        grantedScopes = [parsedScopes copy];
+      }
+
+      if ([GIDScopes containAdditionalScopes:grantedScopes]) {
```

Comment:
There are signIn flow scenarios that will result in grantedScopes containing more than basic profile scopes here, so you'll want another approach to determining whether this is part of an addScopes flow.

Reply:
Add a property in GIDSignInInternalOptions to indicate it is a sign in flow or add scopes flow.